src/core/utils/conversation/hasSufficientRemainingTokens.test.ts

```diff
@@ -0,0 +1,45 @@
+import { describe, expect, it, vi } from 'vitest';
+import { isWithinTokenLimit } from 'gpt-tokenizer';
+import { estimateTokenUsage } from '../../../features/reasoning/helpers';
+import { hasSufficientRemainingTokens } from './hasSufficientRemainingTokens';
+
+describe('hasSufficientRemainingTokens', () => {
+  vi.mock(import('gpt-tokenizer'), async (importOriginal) => {
+    const actual = await importOriginal();
+    return {
+      ...actual,
+      isWithinTokenLimit: vi.fn(),
+      encodeChat: vi.fn(),
+    };
+  });
+
+  vi.mock('../../../features/reasoning/helpers', () => ({
+    estimateTokenUsage: vi.fn(),
+  }));
+
+  afterEach(() => {
+    vi.resetAllMocks();
+  });
+
+  describe('blockchain model', () => {
+    it('calls isWithinTokenLimit with correct arguments', () => {
+      hasSufficientRemainingTokens('gpt-4o', 'test input', 2000);
+
+      expect(isWithinTokenLimit).toHaveBeenCalledWith('test input', 2000);
+      expect(estimateTokenUsage).not.toHaveBeenCalled();
+    });
+  });
```

Comment:
This function simply routes between two helpers, one from a third party app and one of ours that has its own unit tests.